### PR TITLE
Bulk source query flags for thumbnails and nested output

### PR DESCRIPTION
**Pull request by Yashvi-Sharma**
Added a flag "includeThumbnails" to source query, thumbnails are still being returned in bulk source query
Added a flag "removeNested" to remove classifications, annotations and groups output from bulk source query

Closes https://github.com/skyportal/skyportal/issues/1807

### CODE REVIEW

**Comment by acrellin** [comment]
Closes https://github.com/skyportal/skyportal/issues/1807

**Comment by acrellin** [comment]
Thanks @Yashvi-Sharma! I will take a look shortly

**Review by acrellin — COMMENTED** [review]
I'll give a more thorough review soon, but an initial question is below

File: skyportal/handlers/api/source.py
Comment (Yashvi-Sharma) [reply skyportal/handlers/api/source.py:593]: I only put the remove_nested flag for multiple source query. I don't think it is needed for single source queries as they are fast enough

File: skyportal/handlers/api/source.py
Comment (Yashvi-Sharma) [reply skyportal/handlers/api/source.py:778]: Yes I'll add that check

**Comment by acrellin** [comment]
@Yashvi-Sharma Can you please try merging latest master & resolving the conflicts here? Let me know if you want any help with this process. Thanks!

**Comment by acrellin** [comment]
@Yashvi-Sharma  The failures do indeed appear related to the code changes here; the source API response data must've changed. I'll take a look and let you know what I find

**Comment by acrellin** [comment]
@Yashvi-Sharma The front-end API calls will need to be updated to now pass in `includeThumbnails=true`, e.g. https://github.com/skyportal/skyportal/blob/master/static/js/ducks/source.js#L98-L103 and also in the `sources.js` file (there may also be others; you can grep to find all relevant API calls). LMK if you have any questions about how to do that. Thanks!

File: skyportal/handlers/api/source.py
Comment (acrellin) [inline skyportal/handlers/api/source.py:1085]: This whole block needs to be de-indented (not nested data)

Comment (Yashvi-Sharma) [reply skyportal/handlers/api/source.py]: Done

**Comment by Yashvi-Sharma** [comment]
> @Yashvi-Sharma The front-end API calls will need to be updated to now pass in `includeThumbnails=true`, e.g. https://github.com/skyportal/skyportal/blob/master/static/js/ducks/source.js#L98-L103 and also in the `sources.js` file (there may also be others; you can grep to find all relevant API calls). LMK if you have any questions about how to do that. Thanks!

Updated all front-end calls I could find in source.js and sources.js

File: skyportal/tests/api/test_sources.py
Comment (acrellin) [inline skyportal/tests/api/test_sources.py:115]:
```suggestion
    assert all(k not in data["data"]["sources"][0] for k in ["annotations", "groups", "thumbnails"])
```

File: skyportal/tests/api/test_sources.py
Comment (acrellin) [inline skyportal/tests/api/test_sources.py:75]: What's the motivation for creating these sources via the API as opposed to using the fixtures as in the above test?

File: skyportal/tests/api/test_sources.py
Comment (acrellin) [inline skyportal/tests/api/test_sources.py:113]: Shouldn't classifications be considered nested data, too?

**Review by acrellin — COMMENTED** [review]
Looks great so far! Just a few questions below

File: skyportal/tests/api/test_sources.py
Comment (Yashvi-Sharma) [reply skyportal/tests/api/test_sources.py:75]: When I was running the test without creating these sources, the queried list of sources was sometimes empty

File: skyportal/tests/api/test_sources.py
Comment (Yashvi-Sharma) [reply skyportal/tests/api/test_sources.py:113]: Yes but the RCF group needs the query such that it returns all RCF sources with redshift and classifications (the remove nested is needed to speed up the query because for ~8000 sources in RCF it never finishes)

File: skyportal/tests/api/test_sources.py
Comment (Yashvi-Sharma) [reply skyportal/tests/api/test_sources.py:115]: The thumbnails are still being returned for the multiple source query, I don't know why. I can't find where that's happening in the code

**Comment by acrellin** [comment]
@Yashvi-Sharma I pushed up a small commit which allows thumbnails to be successfully excluded

**Comment by Yashvi-Sharma** [comment]
> @Yashvi-Sharma I pushed up a small commit which allows thumbnails to be successfully excluded

Awesome! If everything looks okay, can we merge this PR now?

File: skyportal/handlers/api/source.py
Comment (acrellin) [inline skyportal/handlers/api/source.py:398]:
```suggestion
              Boolean indicating whether to remove nested output (does not apply to classifications). Defaults to false.
```

File: skyportal/tests/api/test_sources.py
Comment (acrellin) [inline skyportal/tests/api/test_sources.py:105]:
```suggestion
        params={"removeNested": True, "group_ids": [public_group.id]},
```

File: skyportal/tests/api/test_sources.py
Comment (acrellin) [inline skyportal/tests/api/test_sources.py:59]:
```suggestion
        params={"includeThumbnails": True},
```

**Review by acrellin — APPROVED** [review]
Looks good, thanks @Yashvi-Sharma! Let's wait for @dmitryduev 's review before merging

**Review by dmitryduev — APPROVED** [review]
Great work, thanks @Yashvi-Sharma!

**Pull request merged**
